_drgn.pyi

```diff
@@ -202,13 +203,27 @@ class Program:
         :attr:`SymbolBinding.WEAK` symbol is found, it is returned. Otherwise,
         any matching symbol (e.g., :attr:`SymbolBinding.LOCAL`) is returned. If
         there are multiple matching symbols with the same binding, one is
-        returned arbitrarily.
+        returned arbitrarily. To retrieve all symbols with this name, regardless
+        of their binding scope, use :meth:`Program.symbols()`.
 
         :param address_or_name: Address or name.
         :raises LookupError: if no symbol contains the given address or matches
             the given name
         """
         ...
+    def symbols(self, name: Optional[str]) -> List[Symbol]:
+        """
+        Get a list of global and local symbols, optionally matching a name.
+
+        When the name parameter is given, this returns all symbols matching the
+        name, unlike :meth:`Program.symbol()`, which returns only one matching
+        symbol. When the name parameter is not provided, this returns a list of
+        all symbols in the program.
+
+        :param name: The symbol name. If not provided, return all symbols.
+        :raises LookupError: if no symbol matches the given name
```

Comment:
Now that this allows returning all symbols in the program, `LookupError` doesn't make as much sense. Maybe we should just return an empty list if nothing matches after all?

Reply:
I think we discussed it on a prior iteration, but I have no particular attachment to the LookupError. I'll return the empty list instead.

Reply:
Right, I think we originally decided on `LookupError` but I think the "all symbols" functionality changes the API enough that it no longer feels right.